src/miner_consensus_mgr.erl

```diff
@@ -474,12 +462,20 @@ handle_info({blockchain_event, {add_block, Hash, Sync, _Ledger}},
                                 case maps:find(EID, State#state.current_dkgs) of
                                     %% we're not in
                                     error ->
-                                        lager:info("stopping old group: election, no key"),
-                                        erlang:demonitor(State2#state.ag_monitor, [flush]),
-                                        stop_group(State2#state.active_group),
-                                        miner:remove_consensus(),
-                                        miner_hbbft_sidecar:set_group(undefined),
-                                        State2#state{active_group = undefined};
+                                        %% if we've never heard of this one, try to restart the DKG,
+                                        %% which we may have stopped
+                                        case restart_dkg(ElectionHeight, ElectionDelay, State2) of
```

Comment:
I don't understand what 'no_dkg' has to do with stopping the hbbft group?

Reply:
the stop is idempotent, it's all noops in the case that we weren't in the group.  it's just in case we restarted right before the block but were not in the new group.